vector/src/main/java/im/vector/app/features/roommemberprofile/RoomMemberProfileViewModel.kt

```diff
@@ -167,9 +169,25 @@ class RoomMemberProfileViewModel @AssistedInject constructor(
             is RoomMemberProfileAction.KickUser               -> handleKickAction(action)
             RoomMemberProfileAction.InviteUser                -> handleInviteAction()
             is RoomMemberProfileAction.SetUserColorOverride   -> handleSetUserColorOverride(action)
+            is RoomMemberProfileAction.OpenOrCreateDm         -> handleOpenOrCreateDm(action)
         }.exhaustive
     }
 
+    private fun handleOpenOrCreateDm(action: RoomMemberProfileAction.OpenOrCreateDm) {
+        viewModelScope.launch {
+            _viewEvents.post(RoomMemberProfileViewEvents.Loading())
+            val roomId = try {
+                directRoomHelper.ensureDMExists(action.userId)
+            } catch (failure: Throwable) {
+                _viewEvents.post(RoomMemberProfileViewEvents.Failure(failure))
+                return@launch
+            }
+            if (roomId != initialState.roomId) {
+                _viewEvents.post(RoomMemberProfileViewEvents.OpenRoom(roomId = roomId))
+            }
+        }
+    }
```

Comment:
This code is duplicate of the code in `TimelineViewModel`.
We should find a way to avoid code duplication in ViewModels (maybe out of scope of this PR)

Reply:
Ah, my bad, sorry